polaris/dataset/zarr/_utils.py

```diff
@@ -11,3 +11,22 @@ def load_zarr_group_to_memory(group: zarr.Group) -> dict:
         elif isinstance(item, zarr.Group):
             data[key] = load_zarr_group_to_memory(item)
     return data
+
+
+def check_zarr_codecs(group: zarr.Group):
+    """Check if all codecs in the Zarr group are registered."""
+    try:
+        for key, item in group.items():
```

Comment:
**Question**
Is this where the exception gets raised, if the codec is not present?

Also,  does this require pulling the underlying data, or does this work purely on the metadata?

Reply:
Yes, it raises the error as soon as you instantiate the array class that uses it. And no, this doesn't pull in any data.